server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -639,6 +645,8 @@ public MemoryUsage overLimit(MemoryUsage memoryUsed) {
                         allocationIndex,
                         allocationDuration
                     );
+                } else {
+                    logger.info("memory usage down, before [{}], after [{}]", memoryUsed.baseUsage, current);
```

Comment:
I think the information here could be combined with the log line above? We could log it here (outside the lock seems best), but capture the info above instead of logging it immediately.